source/server/http/admin.cc

```diff
@@ -539,9 +539,10 @@ Http::Code AdminImpl::handlerConfigDump(absl::string_view, Http::HeaderMap& resp
     auto& any_message = *(dump.add_configs());
     any_message.PackFrom(*message);
   }
+  const auto redacted = MessageUtil::redact(dump);
 
   response_headers.setReferenceContentType(Http::Headers::get().ContentTypeValues.Json);
-  response.add(MessageUtil::getJsonStringFromMessage(dump, true)); // pretty-print
+  response.add(MessageUtil::getJsonStringFromMessage(*redacted, true)); // pretty-print
```

Comment:
I feel we also want this in various logs, e.g. trace level xDS logs, where this has come up previously as an issue.

Reply:
Yes, definitely. I'm thinking to just target this one spot in this PR, as a proof of concept, and then track down others (e.g. #4757) as a follow-up.

Reply:
Sure, I'm good with keeping the issue open for follow-up.